packages/scene-composer/src/components/panels/scene-components/AnimationViewStateEditor.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormField, Button, SpaceBetween, Autosuggest } from '@awsui/components-react';
+import { useIntl } from 'react-intl';
+
+import { IAnimationComponentInternal, useEditorState, ISceneNodeInternal } from '../../../store';
+import { useSceneComposerId } from '../../../common/sceneComposerIdContext';
+import { animationObjectKey } from '../../three-fiber/AnimationComponent/AnimationComponent';
+
+export type AnimationViewStateEditorProps = {
+  node: ISceneNodeInternal;
+  component: IAnimationComponentInternal;
+  onUpdate: (currentAnimations: string[], noderef: string, component: IAnimationComponentInternal) => void;
+};
+export const AnimationViewStateEditor: React.FC<AnimationViewStateEditorProps> = ({
+  node,
+  component: animationComponent,
+  onUpdate,
+}: AnimationViewStateEditorProps) => {
+  const { formatMessage } = useIntl();
+
+  const [panelnumber, setPanelNumber] = useState(0);
+  const sceneComposerId = useSceneComposerId();
+  const { getObject3DBySceneNodeRef } = useEditorState(sceneComposerId);
+  const object = getObject3DBySceneNodeRef(node.ref);
+  //Scene is the name of the object3d of the model as given by the gltfLoader
+  const animationList = object?.getObjectByName(animationObjectKey);
+  const [selectedAnimations, setSelectedAnimations] = React.useState<string[]>();
```

Comment:
so this can be undefined and its type should be `useState<string[] | undefined>`